### DISCUSSION TITLE
Password-less login/signup

### DISCUSSION THREAD

**Discussion by laurentS** [supabase/supabase#505] [Questions]
This is a followup to [another discussion](https://github.com/supabase/supabase/discussions/460#discussioncomment-256759) where a solution to signup with email only was suggested.

I was trying to do a password-less signup form after asking the user for some simple info (you know, trying to make it as low friction as possible :fast_forward: ) but there's an issue:
- if I use `supabase.auth.signUp()` I get the `user_id` in the response, so I can immediately save info in the db that is linked to my new user
- if I use `supabase.auth.signIn()` I'm not getting anything, so I can't do this, which makes me sad :cry: 

It would be cool if the second option was possible. I can't see any major blockers, as the account is created in the backend, it would just need sending back the user data, probably via some different api call, maybe `auth.signUp` but without password? Am I missing something?

**Comment by kiwicopple** [comment]
Hi @laurentS , here are our docs for magic links: https://supabase.io/docs/client/auth-signin#sign-in-with-magic-link

Is there something missing from our docs perhaps?

**Reply by laurentS** [reply]
I was hoping to get my users to sign up without inputting a password right away, so the flow would be a form asking some question, then "input your email" (sends magic link) and at that point, I'd save their answer to the question in their profile (say in `public.user_profiles`). But I don't get the user object back from the `signIn` method, so I can't do this.
Having thought a little more about it, it might make sense from a security perspective, but then I think the docs should mention that `signIn` won't return the user object (looking at the code, the response is something like `{ user: null, ...}`.

**Reply by alz10** [reply]
Have you disable email confirmation?

**Reply by laurentS** [reply]
No, I kept them on.
